src/lib.rs

```diff
@@ -131,11 +162,27 @@ impl Client {
         Self { base_url, client }
     }
 
+    pub(crate) fn build_url(&self, path: &str, params: &[(&str, Option<&str>)]) -> reqwest::Url {
+        // Create a filtered set of string params that were specified.
+        let params = params.iter().filter_map(|p| {
+            if p.1.is_none() {
```

Comment:
this is neat.. There has to be a more idiomatic way to do this though?

Reply:
Sure, happy to iterate here, though some guidance on what direction is most idiomatic would be most welcome! Two possibilities come to mind:

Clarify flow a bit by splitting out the filter/map stages:

```rust
        let params = params
            .iter()
            .filter(|p| p.1.is_some())
            .map(|p| (p.0, p.1.unwrap()));
```

Or this is how I would write it if I was trying to write idiomatic Go code:

```rust
        let mut actual_params = Vec::new();
        for p in params {
            if p.1.is_some() {
                actual_params.push((p.0, p.1.unwrap()));
            }
        }
```

If something else entirely comes to mind let me know.

Reply:
I've pushed the first one as a starting point